tests/main/interfaces-microstack-support/task.yaml

```
@@ -0,0 +1,22 @@
+summary: Ensure that the microstack-support device delegation works.
+
+details: |
+    The microstack-support interface delegates device cgroup handling to the
+    snap.
+
+environment:
+    SNAP_NAME: test-snapd-sh
+prepare: |
+    "$TESTSTOOLS"/snaps-state install-local test-snapd-sh
+
+execute: |
+    echo "When the interface is connected"
+    snap connect "$SNAP_NAME":microstack-support
+
+    snap start "$SNAP_NAME"
+
+    retry -n 50 --wait 0.1 sh -c "test -f /var/snap/$SNAP_NAME/common/proc_self_cgroup"
+    CGROUP=$(grep ':devices:' "/var/snap/$SNAP_NAME/common/proc_self_cgroup" | cut -d':' -f3)
+    echo "Service chroup is $CGROUP"
+
+    cat "/sys/fs/cgroup/devices/$CGROUP/devices.list" | MATCH 'a \*:\* rwm'
```

shellcheck is silly and thinks this is better:

```suggestion
    MATCH 'a \*:\* rwm' < "/sys/fs/cgroup/devices/$CGROUP/devices.list"
```